packages/store/addon/-private/system/core-store.ts

```diff
@@ -2657,7 +2658,12 @@ abstract class CoreStore extends Service {
     internalModel.setupData(data);
 
     if (!isUpdate) {
-      this.recordArrayManager.recordDidChange(internalModel);
+      // TODO: do we want to maintain old API with this flag?
+      if (RECORD_ARRAY_MANAGER_IDENTIFIERS) {
+        this.recordArrayManager.recordDidChange(internalModel.identifier);
```

Comment:
Given that we use the identifier on line 2644, we could extract the `let identifier`